sdk/src/main/java/com/hedera/hashgraph/sdk/TransactionRecordQuery.java

```diff
@@ -115,25 +115,13 @@ ExecutionState shouldRetry(Status status, Response response) {
             case RECORD_NOT_FOUND:
                 return ExecutionState.Retry;
 
-            case SUCCESS:
-            case IDENTICAL_SCHEDULE_ALREADY_CREATED:
-                return ExecutionState.Finished;
-
             default:
-                return ExecutionState.Error;
+                return ExecutionState.Finished;
         }
     }
 
     @Override
-    Exception mapStatusError(Status status, @Nullable TransactionId transactionId, Response response) {
-        if (status != Status.OK) {
-            return new PrecheckStatusException(status, transactionId);
-        }
-
-        // has reached consensus but not generated
-        return new ReceiptStatusException(
-            Objects.requireNonNull(transactionId),
-            TransactionReceipt.fromProtobuf(response.getTransactionGetRecord().getTransactionRecord().getReceipt())
-        );
+    PrecheckStatusException mapStatusError(Status status, @Nullable TransactionId transactionId, Response response) {
```

Comment:
This override of `mapStatusError()` can be removed, since it is no different from the default implementation in `Executable`